use-cases.md

```diff
@@ -69,11 +65,7 @@ The `operation` above refers to the name the server wants to give to the Span. F
 The `span` object above can be `None` if the Tracer did not find relevant headers in the incoming request: presumably because the client did not send them. In this case the server needs to start a brand new trace.
 
 ```python
-    span = tracer.join_trace_from_text(
-        operation_name=operation,
-        context_snapshot=request.headers,
-        trace_attributes=request.headers
-    )
+    span = tracer.get_span_extractor(request.class).extract(request)
```

Comment:
After sleeping on this, I would prefer that we use something more explicit than param-type-based inference to determine the appropriate injector/extractor for the given job. This could be via some sort of registry mechanism or it could be disgusting enum. But using the type alone seems fragile and "magical in a bad way."

I still like the core premise of this PR: namely, an "official" interface for injection/extraction (rather than the two ad hoc `SpanPropagator` methods we have right now) as it makes it easier to imagine adding support for particularly popular protocols at the OT layer... I'm thinking of large, polyglot, popular IPC mechanisms like grpc, thrift, tchannel, whatever.

@michaelsembwever it would be great to talk about this at our inaugural hangout this morning/afternoon/evening depending on your TZ :)